spec/src/language_negotiation_lambda_spec.js

```diff
@@ -0,0 +1,258 @@
+'use strict';
+
+var languageNegotiationLambda = require( '../../src/language_negotiation_lambda' );
+
+describe( 'languageNegotiationLambda', function() {
+    var event = {
+        "Records": [
+            {
+              "cf": {
+                "request": {
+                  "headers": {
+                    "Host": [
+                      "localhost"
+                    ],
+                    "User-Agent": [
+                      "Test Agent"
+                    ]
+                  },
+                  "clientIp": "2001:cdba::3257:9652",
+                  "httpVersion": "2.0",
+                  "uri": "/",
+                  "method": "GET"
+                },
+                "config": {
+                  "distributionId": "EXAMPLE"
+                }
+              }
+            }
+        ]
+    };
+    var default_header = ['en'];
+
+    beforeEach(function(){
+        delete event.Records[0].cf.request.headers["Cookie"];
+        delete event.Records[0].cf.request.headers["Accept-Language"];
+        delete event.Records[0].cf.request.headers["X-Accept-Language"];
+    });
+
+    describe( 'Well formed cookies', function(){
+        [
+            { cookie: "en",     expected: "en" },
+            { cookie: "EN",     expected: "en" },
+            { cookie: "es",     expected: "es" },
+            { cookie: "ES",     expected: "es" },
+            { cookie: "en-US",  expected: "en" },
+            { cookie: "es-419", expected: "es" }
+        ].forEach( function( cookies ) {
+            it( `Resolves cookie ${cookies.cookie} to ${cookies.expected}`, function( done ) {
+                event.Records[0].cf.request.headers.Cookie = [
+                    process.env.LANGUAGE_COOKIE_NAME+"="+cookies.cookie
```

Comment:
Nitpick: Not sure if it's in our style guide or not, but I think this would be easier to read with whitespace before/after the `+`.